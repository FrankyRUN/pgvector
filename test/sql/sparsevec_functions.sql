SELECT l2_distance('{}/2'::sparsevec, '{0:3,1:4}/2');
SELECT l2_distance('{}/2'::sparsevec, '{1:1}/2');
SELECT '{}/2'::sparsevec <-> '{0:3,1:4}/2';

SELECT inner_product('{0:1,1:2}/2'::sparsevec, '{0:2,1:4}/2');
SELECT sparsevec_negative_inner_product('{0:1,1:2}/2', '{0:2,1:4}/2');

SELECT cosine_distance('{0:1,1:2}/2'::sparsevec, '{0:2,1:4}/2');
SELECT cosine_distance('{0:1,1:2}/2'::sparsevec, '{}/2');
SELECT cosine_distance('{0:1,1:1}/2'::sparsevec, '{0:-1,1:-1}/2');
SELECT cosine_distance('{0:1}/2'::sparsevec, '{1:2}/2');
SELECT cosine_distance('{}/1'::sparsevec, '{}/1');
SELECT cosine_distance('{0:1}/2'::sparsevec, '{0:1}/3');
