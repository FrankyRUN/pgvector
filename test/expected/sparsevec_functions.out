SELECT l2_distance('{}/2'::sparsevec, '{0:3,1:4}/2');
 l2_distance 
-------------
           5
(1 row)

SELECT l2_distance('{}/2'::sparsevec, '{1:1}/2');
 l2_distance 
-------------
           1
(1 row)

SELECT '{}/2'::sparsevec <-> '{0:3,1:4}/2';
 ?column? 
----------
        5
(1 row)

SELECT inner_product('{0:1,1:2}/2'::sparsevec, '{0:2,1:4}/2');
 inner_product 
---------------
            10
(1 row)

SELECT sparsevec_negative_inner_product('{0:1,1:2}/2', '{0:2,1:4}/2');
 sparsevec_negative_inner_product 
----------------------------------
                              -10
(1 row)

SELECT cosine_distance('{0:1,1:2}/2'::sparsevec, '{0:2,1:4}/2');
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('{0:1,1:2}/2'::sparsevec, '{}/2');
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('{0:1,1:1}/2'::sparsevec, '{0:-1,1:-1}/2');
 cosine_distance 
-----------------
               2
(1 row)

SELECT cosine_distance('{0:1}/2'::sparsevec, '{1:2}/2');
 cosine_distance 
-----------------
               1
(1 row)

SELECT cosine_distance('{}/1'::sparsevec, '{}/1');
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('{0:1}/2'::sparsevec, '{0:1}/3');
ERROR:  different sparsevec dimensions 2 and 3
